src/request/TopLevelApi.js

```diff
@@ -125,31 +125,49 @@ class TopLevelApi extends RequestParser { // eslint-disable-line no-unused-vars
                 return false;
             });
 
-            this.onRequest(parsedRequest).catch(reject);
-            this.setGlobalCloseButtonText(`${I18n.translatePhrase('back-to')} ${parsedRequest.appName}`);
-            TopLevelApi.setLoading(false);
+            if (!this.Handler) {
+                reject(new Errors.KeyguardError('Handler undefined'));
+                return;
+            }
+            try {
+                const handler = new this.Handler(parsedRequest, resolve, reject);
+
+                this.onBeforeRun(parsedRequest);
+
+                this.setGlobalCloseButtonText(`${I18n.translatePhrase('back-to')} ${parsedRequest.appName}`);
+
+                TopLevelApi.setLoading(false);
+
+                handler.run();
```

Comment:
Isn't this supposed to be the other way around?
Hander.run() zuerst und danach setLoading(false)?